arch-snapshot/arch.md

```diff
@@ -48,9 +48,300 @@ only as detailed as to explain how the actors control the artifacts. The goal is
+declared in Agreement a specific amount of RAM and a certain number of threads to be allocated for the VM.
+Provider can only sell any remaining RAM and cores to other Requestors.
+
+Resource access always occurs within the context of an Activity. Creating an Activity is synonymous with starting an
+execution environment. The Activity module enables the Requestor to control the execution environment and monitor
+its state. Further details on controlling an Activity from the Requestor's perspective can be found in
+the ["Running something"](#running-something) section.
+
+From the Provider's perspective, the primary focus is to listen for incoming Activity events and create an Activity
+when requested by the Requestor. Upon receiving an Activity creation event, the Provider should spawn an ExeUnit process
+(and a Virtual Machine in consequence). Conversely, receiving an Activity destruction event should trigger the termination
+of the ExeUnit processes.
+
+The Requestor is allowed to spawn multiple Activities consecutively. In general, multiple Activities running
+simultaneously may be permitted; however, this does not apply in the case of a Virtual Machine, as hardware
+resources can only be allocated once.
+
+#### 5. Send DebitNotes to notify Requestor of updated costs
+
+The ExeUnit is directly controlled by the Requestor Agent, with no intervention from the Provider. Communication
+happens solely between the Yagna daemon and the ExeUnit process. The Provider's responsibility is limited to calculating
+the cost of resource usage based on the pricing model defined in the Agreement and informing the Requestor accordingly.
+The ExeUnit tracks resource consumption, while the Provider Agent computes and communicates the cost to the Requestor
+via Debit Notes. The Provider Agent must also monitor the acceptance of these Debit Notes, as it signifies
+the Requestor's commitment to pay the specified amount.
+
+Debit Notes are formal documents used to notify the Requestor of the costs incurred. Each note details the amount owed 
+for a specific activity up to the time the note is issued. As each subsequent Debit Note reflects the updated costs,
+it effectively supersedes the previous one.
+
+The frequency of sending Debit Notes and the acceptance deadlines are outlined in the Agreement. It is important to note
+that accepting a Debit Note does not trigger immediate payment. Debit Notes can be classified as payable or non-payable,
+with payable Debit Notes identified by the due date included in the document. While payable Debit Notes are scheduled
+for processing upon acceptance, this does not necessitate immediate payment. The payment mechanism allows for the batching
+of payments or delaying them to accommodate additional Debit Notes or Invoices, thereby reducing transaction costs on
+the blockchain. Consequently, while payments are not immediate, they must be completed before the due date specified
+in the Agreement.
+
+```mermaid
+sequenceDiagram
```

Comment:
Similarly to the previous one, this diagram is great but I'd rather save it for the technical view. We do not need to introduce the user to individual processes in this chapter.

Reply:
I placed the original diagrams in chapters that will talk diving into more details. I will just leave them there, let's work on them later.